pkg/segment/query/iqr/intermediateQueryResult.go

```diff
@@ -743,10 +746,45 @@ func (iqr *IQR) RenameColumn(oldName, newName string) error {
 		return err
 	}
 
-	iqr.renamedColumns[oldName] = newName
-	if values, ok := iqr.knownValues[oldName]; ok {
+	// delete newName since it would be overwritten
+	delete(iqr.knownValues, newName)
```

Comment:
I don't understand why you replaced line 746 with line 750. Can you explain it to me?

Reply:
It was just to separate things out to easily get the flow.
- If we rename `fieldA` to `fieldB`, it means that `fieldB` if present needs to be deleted.
- If `fieldA` was a created/read column, we can just populate it.
- If not, we need to update the chain for renaming since, it could be a column in RRC. There are 2 cases:
  - Was old name previously renamed to something if so mark it as renamed to the latest newname.
  - If not, just add the rename mapping to the renamed columns.